### PR TITLE
Add VTI functionality

### PR DESCRIPTION
**Pull request by hsychla**
Add functionality to create a VTI tunnel interface for easier policy- and routing-changes.

### CODE REVIEW

File: files/start-strongswan.sh [hunk 1]
```diff
@@ -1,20 +1,59 @@
...
+_create_vti() {
+
+# set charon.install_virtual_ip = no to prevent the daemon from also installing the VIP
+
+    if [ -n "$IPSEC_VTI_KEY" ]
+    echo "IPSEC_VTI_KEY set, creating VTI interface."
+    then
+        set -o nounset
+        set -e
+        set -x
```
Comment (thz) [inline files/start-strongswan.sh]: please remove -x or set it just for some DEBUG mode.

File: files/start-strongswan.sh [hunk 2]
```diff
@@ -1,20 +1,59 @@
...
+
+        echo "Start: load ip_vti kernel module."
+        modprobe ip_vti
```
Comment (thz) [inline files/start-strongswan.sh]: can you probe if the loading is necessary?

File: files/start-strongswan.sh [hunk 3]
```diff
@@ -49,25 +88,37 @@ _term() {
 }
 
 _set_default_variables() {
-    export IPSEC_REMOTEIP=${IPSEC_REMOTEIP:-%any}
-    export IPSEC_LOCALIP=${IPSEC_LOCALIP:-%any}
+    # local and remote IP can not be "%any" if VTI needs to be created
+    if [ -z $IPSEC_VTI_KEY ]
```
Comment (thz) [inline files/start-strongswan.sh]: please add quotes. (it is legal when the var is empty, yes - but it becomes problematic with invalid values)

File: files/start-strongswan.sh [hunk 4]
```diff
@@ -49,25 +88,37 @@ _term() {
...
-  [ -z "$IPSEC_REMOTEIP" ] && { echo "Need to set IPSEC_REMOTEIP"; exit 1; }
-  [ -z "$IPSEC_REMOTEID" ] && { echo "Need to set IPSEC_REMOTEID"; exit 1; }
-  [ -z "$IPSEC_LOCALIP" ] && { echo "Need to set IPSEC_LOCALIP"; exit 1; }
-  [ -z "$IPSEC_LOCALID" ] && { echo "Need to set IPSEC_LOCALID"; exit 1; }
-  [ -z "$IPSEC_REMOTENET" ] && { echo "Need to set IPSEC_REMOTENET"; exit 1; }
-  [ -z "$IPSEC_KEYEXCHANGE" ] && { echo "Need to set IPSEC_KEYEXCHANGE"; exit 1; }
-  [ -z "$IPSEC_ESPCIPHER" ] && { echo "Need to set IPSEC_ESPCIPHER"; exit 1; }
-  [ -z "$IPSEC_IKECIPHER" ] && { echo "Need to set IPSEC_IKECIPHER"; exit 1; }
+  # we only need two varaiables for init-containers
+  if [ -n $IPSEC_VTI_KEY ]
```
Comment (thz) [inline files/start-strongswan.sh]: quotes please.

File: files/start-strongswan.sh [hunk 5]
```diff
@@ -83,6 +134,7 @@ _print_variables() {
     printf "IPSEC_KEYEXCHANGE=%s\n" $IPSEC_KEYEXCHANGE
     printf "IPSEC_ESPCIPHER=%s\n" $IPSEC_ESPCIPHER
     printf "IPSEC_IKECIPHER=%s\n" $IPSEC_IKECIPHER
+    printf "IPSEC_VTI_KEY=%s\n" $IPSEC_VTI_KEY
```
Comment (thz) [inline files/start-strongswan.sh:136]: IPSEC_VTI_STATICROUTES output missing?

File: README.md [hunk 6]
```diff
@@ -97,3 +97,22 @@ IPSEC_FORCEUDP=
...
+* IPSEC_REMOTEIP
+* IPSEC_VTI_KEY
+* IPSEC_VTI_STATICROUTES
+
+If the entrypoint is provided the argument `init`, an initialisation-container is started that can create a VTI tunnel to route
+the IPSEC traffic over.
+
+To create a VTI interface, set the environment variable `IPSEC_VTI_KEY` to an integer.
+
+A VTI tunnel interface is then created with `IPSEC_LOCALIP` and `IPSEC_REMOTEIP` as endpoints and `IPSEC_VTI_KEY` as key.
```
Comment (thz) [inline README.md]: is that still / really necessary to have the peer IPs on the vti?

**Review by thz — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (hsychla) [reply README.md]: Test make it look like no: 

```
14: vti42@NONE: <NOARP,UP,LOWER_UP> mtu 1332 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1
    link/ipip 92.43.50.69 brd 0.0.0.0
```

```
vti42: ip/ip remote any local 92.43.50.69 ttl inherit key 42
RX: Packets    Bytes        Errors CsumErrs OutOfSeq Mcasts
    122        10248        0      0        0        0
TX: Packets    Bytes        Errors DeadLoop NoRoute  NoBufs
    122        19764        0      0        0        0
```

I removed it.

File: files/start-strongswan.sh (hunk 1 above)
Comment (hsychla) [reply files/start-strongswan.sh]: done

File: files/start-strongswan.sh (hunk 2 above)
Comment (hsychla) [reply files/start-strongswan.sh]: done

File: files/start-strongswan.sh (hunk 3 above)
Comment (hsychla) [reply files/start-strongswan.sh]: done

File: files/start-strongswan.sh (hunk 4 above)
Comment (hsychla) [reply files/start-strongswan.sh]: done

File: files/start-strongswan.sh (hunk 5 above)
Comment (hsychla) [reply files/start-strongswan.sh:136]: yes, added.

**Review by thz — APPROVED** [review]

**Pull request merged**